flink-cdc-runtime/src/main/java/com/ververica/cdc/runtime/operators/schema/coordinator/SchemaManager.java

```diff
@@ -173,27 +167,8 @@ private void handleAlterColumnTypeEvent(AlterColumnTypeEvent event) {
                 event.tableId());
 
         LOG.info("Handling schema change event: {}", event);
-        Map<String, DataType> typeMapping = event.getTypeMapping();
-        Schema oldSchema = optionalSchema.get();
-        Schema.Builder schemaBuilder = Schema.newBuilder();
-
-        // Rebuild physical columns
-        for (Column column : oldSchema.getColumns()) {
-            if (typeMapping.containsKey(column.getName())) {
-                // The column type is being changed
-                Column newColumn = column.copy(typeMapping.get(column.getName()));
-                schemaBuilder.column(newColumn);
-            } else {
-                schemaBuilder.column(column);
-            }
-        }
-
-        // Dump the rest of information
-        schemaBuilder.primaryKey(oldSchema.primaryKeys());
-        schemaBuilder.comment(oldSchema.comment());
-        schemaBuilder.options(oldSchema.options());
-        Schema newSchema = schemaBuilder.build();
-        registerNewSchema(event.tableId(), newSchema);
+        registerNewSchema(
```

Comment:
We can remove the if-else-if tree in `SchemaManager#applySchemaChange`, as `SchemaUtils#applySchemaChangeEvent` does the `instanceof` logic too.